packages/webdriver-utils/test/metadata/normalizeData.test.js

```diff
@@ -0,0 +1,84 @@
+import NormalizeData from '../../src/metadata/normalizeData.js';
+
+describe('NormalizeData', () => {
+  let normalizeDataObj;
+  beforeAll(() => {
+    normalizeDataObj = new NormalizeData();
+  });
+
+  afterAll(() => {
+    normalizeDataObj = null;
+  });
+
+  describe('osRollUp', () => {
+    it('should rollup windows os accordingly', () => {
+      expect(normalizeDataObj.osRollUp('win8.1')).toBe('Windows');
+      expect(normalizeDataObj.osRollUp('win8')).toBe('Windows');
+      expect(normalizeDataObj.osRollUp('win7')).toBe('Windows');
+      expect(normalizeDataObj.osRollUp('win11')).toBe('Windows');
+      expect(normalizeDataObj.osRollUp('Win10')).toBe('Windows');
+      expect(normalizeDataObj.osRollUp('Windows')).toBe('Windows');
+    });
+
+    it('should rollup mac os accordingly', () => {
+      expect(normalizeDataObj.osRollUp('mac')).toBe('OS X');
+      expect(normalizeDataObj.osRollUp('macos')).toBe('OS X');
+      expect(normalizeDataObj.osRollUp('Mac')).toBe('OS X');
+    });
+
+    it('should rollup iphone os accordingly', () => {
+      expect(normalizeDataObj.osRollUp('iphone')).toBe('iOS');
+      expect(normalizeDataObj.osRollUp('iphone8.1')).toBe('iOS');
+      expect(normalizeDataObj.osRollUp('ios')).toBe('iOS');
+    });
+
+    it('should rollup android os accordingly', () => {
+      expect(normalizeDataObj.osRollUp('android')).toBe('Android');
+      expect(normalizeDataObj.osRollUp('android8.1')).toBe('Android');
+    });
+
+    it('should return any other os as it is', () => {
+      expect(normalizeDataObj.osRollUp('fireOS')).toBe('fireOS');
+    });
+  });
+
+  describe('browserRollUp', () => {
+    describe('should rollup browsers for devices', () => {
+      it('should roll up android to chrome', () => {
+        expect(normalizeDataObj.browserRollUp('chrome_android', true)).toBe('chrome');
+      });
+
+      it('should roll up iPhone/iPad to safari', () => {
+        expect(normalizeDataObj.browserRollUp('iPhone', true)).toBe('safari');
+        expect(normalizeDataObj.browserRollUp('iPad', true)).toBe('safari');
+      });
+
+      it('should return default if no condition is matched', () => {
+        expect(normalizeDataObj.browserRollUp('abc', true)).toBe('abc');
+      });
+    });
+
+    describe('should return browsers as it is for desktop', () => {
+      it('should return the browser as detected', () => {
+        expect(normalizeDataObj.browserRollUp('Chrome', false)).toBe('Chrome');
+      });
+    });
+  });
+
+  describe('browserVersionRollUp', () => {
+    describe('should rollup browser version for devices', () => {
```

Comment:
Maybe we should rename the function ? or use two separate here ?

Reply:
Didn't get it properly? Open for new function rename as well. As of now I kept it as for devices `browserVersion` will be deviceName (initially we were thinking we will show OS version, however things didn't seemed consistent on UI), and for devices we are showing the actual `browserVersion`.